docs/dev/opensearch-same-table-join-query.md

```diff
@@ -0,0 +1,1368 @@
+needs to throw for the user to correct the syntax.
+
+The results of this example are exactly the same as case 4.1, but only works with a mapping that is simplified.  The following
+mapping change is needed from the demo mapping provided in annex A. 
+
+Mapping data:
+```json
+      "house_relation": {
+        "type": "join",
+        "relations": {
+          "house": "member"
+        }
+      },
+```
+
+Example query (SQL):
+```sql
+SELECT *
+FROM got AS m
+JOIN got as h USING house_relation
+WHERE h.house.name = "Targaryan"
+```
+
+Expected Request to OpenSearch:
+```json
+{
+    "query": {
+        "bool": {
+            "must": [
+                {
+                    "has_parent": {
+                        "parent_type": "house",
+                        "query": {
+                            "match": {
+                                "house.name": "Targaryen"
+                            }
+                        }
+                    }
+                },
+                {
+                    "term": {
+                        "house_relation": "member"
+                    }
+                }
+            ]
+        }
+    },
+    "_source": {
+        "includes": [],
+        "excludes": []
+    }
+}
+```
+
+Expected Response (OpenSearch-JSON):
+```json
+{
+    "took": 5,
+    "timed_out": false,
+    "_shards": {
+        "total": 1,
+        "successful": 1,
+        "skipped": 0,
+        "failed": 0
+    },
+    "hits": {
+        "total": {
+            "value": 1,
+            "relation": "eq"
+        },
+        "max_score": 1.0,
+        "hits": [
+            {
+                "_index": "got-joins",
+                "_id": "4",
+                "_score": 1.0,
+                "_routing": "1",
+                "_source": {
+                    "house_relation": {
+                        "name": "member",
+                        "parent": "1"
+                    },
+                    "name": {
+                        "firstname": "Daenerys",
+                        "lastname": "Targaryen"
+                    },
+                    "nickname": "Daenerys \"Stormborn\"",
+                    "gender": "F",
+                    "parents": {
+                        "father": "Aerys",
+                        "mother": "Rhaella"
+                    },
+                    "titles": [
+                        {
+                            "title": "Mother Of Dragons"
+                        },
+                        {
+                            "title": "Queen Of The Andals"
+                        },
+                        {
+                            "title": "Breaker Of Chains"
+                        },
+                        {
+                            "title": "Khaleesi"
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+```
+
+### 4.3 Case SELECT... FROM... JOIN ON... WHERE parent relation
+
+Summary:
```

Comment:
Filling in at least a small overview for each of these would examples would be useful.